src/senaite/core/content/labproduct.py

```diff
@@ -0,0 +1,226 @@
+# -*- coding: utf-8 -*-
+#
+# This file is part of SENAITE.CORE.
+#
+# SENAITE.CORE is free software: you can redistribute it and/or modify it under
+# the terms of the GNU General Public License as published by the Free Software
+# Foundation, version 2.
+#
+# This program is distributed in the hope that it will be useful, but WITHOUT
+# ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
+# FOR A PARTICULAR PURPOSE. See the GNU General Public License for more
+# details.
+#
+# You should have received a copy of the GNU General Public License along with
+# this program; if not, write to the Free Software Foundation, Inc., 51
+# Franklin Street, Fifth Floor, Boston, MA 02110-1301 USA.
+#
+# Copyright 2018-2024 by it's authors.
+# Some rights reserved, see README and LICENSE.
+
+from decimal import Decimal
+
+from AccessControl import ClassSecurityInfo
+from ComputedAttribute import ComputedAttribute
+from bika.lims import api
+from bika.lims import senaiteMessageFactory as _
+from bika.lims.interfaces import IDeactivable
+from plone.supermodel import model
+from plone.autoform import directives
+from Products.CMFCore import permissions
+from senaite.core.catalog import SETUP_CATALOG
+from senaite.core.content.base import Item
+from senaite.core.interfaces import IHavePrice
+from senaite.core.interfaces import ILabProduct
+from z3c.form.interfaces import IEditForm
+from zope import schema
+from zope.interface import implementer
+from zope.interface import provider
+from zope.schema.interfaces import IContextAwareDefaultFactory
+
+
+@provider(IContextAwareDefaultFactory)
+def default_vat_factory(context):
+    """Returns the default body text for publication emails
+    """
+    defaultVAT = api.get_setup().getVAT() or "0.00"
+    return Decimal(defaultVAT)
+
+
+class ILabProductSchema(model.Schema):
+    """LabProduct Schema interface
+    """
+
+    title = schema.TextLine(
+        title=_(
+            "title_labproduct_title",
+            default="Name"
+        ),
+        required=True,
+    )
+
+    description = schema.Text(
+        title=_(
+            "title_labproduct_description",
+            default="Description"
+        ),
+        required=False,
+    )
+
+    labproduct_volume = schema.TextLine(
+        title=_(
+            "title_labproduct_volume",
+            default="Volume"
+        ),
+        required=False,
+    )
+
+    labproduct_unit = schema.TextLine(
+        title=_(
+            "title_labproduct_unit",
+            default="Unit"
+        ),
+        required=False,
+    )
+
+    labproduct_price = schema.Decimal(
+        title=_(
+            "title_labproduct_price",
+            default="Price (excluding VAT)"
+        ),
+        description=_(
+            "description_labproduct_price",
+            default="Please provide the price excluding VAT"
+        ),
+        required=True,
+    )
+
+    labproduct_vat = schema.Decimal(
```

Comment:
We want to hold all numeric values as strings to avoid any roundings.
Therefore, you must do the following here:

```python
directives.widget("labproduct_vat", klass="numeric")
labproduct_vat = schema.TextLine(...)
```